src/include/common/code_utils.hpp

```diff
@@ -109,14 +109,17 @@
  * @param[in]  aCondition  A Boolean expression to be evaluated.
  * @param[in]  aAction     An optional expression or block to execute when the assertion fails.
  */
-#define VerifyOrExit(...)                 \
-    do                                    \
-    {                                     \
-        if (!(OT_FIRST_ARG(__VA_ARGS__))) \
-        {                                 \
-            OT_SECOND_ARG(__VA_ARGS__);   \
-            goto exit;                    \
-        }                                 \
+#define VerifyOrExit(...)               \
+    do                                  \
+    {                                   \
+        if (OT_FIRST_ARG(__VA_ARGS__))  \
+        {                               \
+        }                               \
+        else                            \
+        {                               \
+            OT_SECOND_ARG(__VA_ARGS__); \
+            goto exit;                  \
+        }                               \
```

Comment:
I think you initially used `break` from `while()` loop?  The `if()/else` model seems much nicer. 👍

Reply:
yes, having the 'break' may result in an additional 'nop' instruction under some architecture.